app/main.py

```diff
@@ -1,13 +1,22 @@
-def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
-
-
-def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
-
-
-def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+def format_linter_error(error: dict) -> dict:
+    return {
+        "line": error["line_number"], "column": error["column_number"],
+        "message": error["text"], "name": error["code"], "source": "flake8"
+    }
+
+
+def format_single_linter_file(file_path: str, errors: list) -> dict:
+    return {
+        "errors": [format_linter_error(error) for error in errors],
+        "path": file_path,
+        "status": "failed" if errors else "passed"
+    }
+
+
+def format_linter_report(linter_report: dict) -> list:
+    return [
+        {"errors": [], "path": path, "status": "passed"}
+        if not report
+        else format_single_linter_file(path, report)

```

Comment:
You still are not reusing your code, now your `format_single_linter_file` is handling both cases if passed or not, this `if` is redundant